core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -225,22 +228,36 @@ object RequestChannel extends Logging {
 
   }
 
-  /** responseAsString should only be defined if request logging is enabled */
-  class Response(val request: Request, val responseSend: Option[Send], val responseAction: ResponseAction,
-                 val responseAsString: Option[String]) {
-    request.responseCompleteTimeNanos = Time.SYSTEM.nanoseconds
-    if (request.apiLocalCompleteTimeNanos == -1L) request.apiLocalCompleteTimeNanos = Time.SYSTEM.nanoseconds
+  abstract class Response(val request: Request) {
+    val nowNs = Time.SYSTEM.nanoseconds
```

Comment:
We could probably do this in a `locally` block since we don't need `nowNs` except to initialize the fields below.